internal/api/preview_config.go

```diff
@@ -1,19 +1,26 @@
 package api
 
 type PreviewConfig struct {
-	ProjectSlug   string                 `json:"projectSlug"`
-	Credentials   map[string]string      `json:"credentials"`
-	PackageParams map[string]interface{} `json:"packageParams"`
+	ProjectSlug string                    `json:"projectSlug"`
```

Comment:
We don't need ProjectSlug, the project ID is in the query already.

Reply:
It is needed, this is the definition of the JSON file and the source of the slug/id when sending the request over.

Reply:
this is what the mutation currently looks like:

```
mutation deployPreviewEnvironment($organizationId: ID!, $projectId: ID!, $packageConfigurations: JSON!, $ciContext: JSON!, $credentials: JSON!) {
        deployPreviewEnvironment(
          organizationId: $organizationId,
          projectId: $projectId,
          input: {
            credentials: $credentials
            packageConfigurations: $packageConfigurations,
            ciContext: $ciContext,
          }
          ) {
          result {
            name
            slug
            mode
          }
        }
      }
      ```
      
 We currently have the field in there as `projectId`.